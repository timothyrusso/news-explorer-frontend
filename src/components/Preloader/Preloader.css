.preloader {
  margin: 0;
  width: 100%;
  min-height: 282px;
  background-color: #F5F6F7;
  display: flex;
  flex-direction: column;
}

.circle-preloader {
  display: block;
  width: 74px;
  height: 74px;
  border: 1px solid rgb(113, 113, 113);
  border-bottom-color: rgb(130, 126, 126);
  border-left-color: rgb(167, 165, 165);
  border-top-color: rgb(229, 228, 228);
  border-radius: 50%;

  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: 80px auto 24px;

  animation: spin .55s infinite linear;
}

@keyframes spin {
  100% {
    transform: rotate(360deg);
  }
}

.preloader__text {
  text-align: center;
  margin: 0;
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 1.33;
  color: #B6BCBF;
}
