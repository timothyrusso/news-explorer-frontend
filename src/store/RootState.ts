import { UserState } from './user/user.reducer';
import { ArticleState } from './article/article.reducer';
import { TogglesState } from './toggles/toggles.reducer';
import { ErrorsState } from './errors/errors.reducer';

export type RootState = {
  user: UserState;
  article: ArticleState;
  toggles: TogglesState;
  errors: ErrorsState;
};
