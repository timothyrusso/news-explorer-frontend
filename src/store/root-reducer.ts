import { combineReducers } from 'redux';
import { userReducer } from './user/user.reducer';
import { articleReducer } from './article/article.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  article: articleReducer,
});
