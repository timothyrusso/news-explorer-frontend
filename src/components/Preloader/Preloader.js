import "./Preloader.css";

const Preloader = () => {
  return (
    <div className="preloader">
      <i className="circle-preloader"></i>
      <p className="preloader__text">Searching for news...</p>
    </div>
  );
};

export default Preloader;
