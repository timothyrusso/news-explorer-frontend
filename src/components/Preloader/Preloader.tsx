import React, { FC } from 'react';
import './Preloader.css';

type PreloaderProps = {
  text?: string;
};

const Preloader: FC<PreloaderProps> = ({ text = 'Searching for news...' }) => {
  return (
    <div className="preloader" aria-live="polite">
      <i className="circle-preloader"></i>
      <p className="preloader__text">{text}</p>
    </div>
  );
};

export default Preloader;
