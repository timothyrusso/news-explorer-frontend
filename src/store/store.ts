import { legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from './root-reducer';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { RootState } from './RootState';

const store = createStore(rootReducer, applyMiddleware(thunk));

export type AppDispatch = ThunkDispatch<RootState, unknown, AnyAction>;

export default store;
