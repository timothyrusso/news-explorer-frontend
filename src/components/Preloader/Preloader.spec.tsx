import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store/store';
import Preloader from './Preloader';

describe('Preloader component', () => {
  it('Should render the Preloader component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Preloader />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
