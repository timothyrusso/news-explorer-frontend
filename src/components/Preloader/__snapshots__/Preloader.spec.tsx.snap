// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component Should render the Preloader component 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="preloader"
  >
    <i
      class="circle-preloader"
    />
    <p
      class="preloader__text"
    >
      Searching for news...
    </p>
  </div>
</DocumentFragment>
`;
